indexing-hadoop/src/main/java/org/apache/druid/indexer/JobHelper.java

```diff
@@ -343,11 +338,18 @@ public static void injectDruidProperties(Configuration configuration, List<Strin
     }
   }
 
-  public static Configuration injectSystemProperties(Configuration conf)
+  public static Configuration injectSystemProperties(Configuration conf, List<String> listOfAllowedPrefix)
   {
     for (String propName : HadoopDruidIndexerConfig.PROPERTIES.stringPropertyNames()) {
       if (propName.startsWith("hadoop.")) {
         conf.set(propName.substring("hadoop.".length()), HadoopDruidIndexerConfig.PROPERTIES.getProperty(propName));
+      } else {
+        for (String prefix : listOfAllowedPrefix) {
+          if (propName.equals(prefix) || propName.startsWith(prefix + ".")) {
+            conf.set(propName, HadoopDruidIndexerConfig.PROPERTIES.getProperty(propName));
+            break;
+          }
+        }
```

Comment:
`injectDruidProperties` has an almost same block with this. How about adding a new method `HadoopDruidIndexerConfig.getAllowedProperties()` which returns a map of a property name to a property? And this new method can be used both in `injectDruidProperties()` and this method.

Reply:
Done